x-pack/plugins/ml/public/application/components/data_grid/use_column_chart.ts

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import moment from 'moment';
+import { BehaviorSubject } from 'rxjs';
+
+import { useObservable } from 'react-use';
+
+import { euiPaletteColorBlind, EuiDataGridColumn } from '@elastic/eui';
+
+import { KBN_FIELD_TYPES } from '../../../../../../../src/plugins/data/public';
+
+import { stringHash } from '../../../../common/util/string_utils';
+
+import { NON_AGGREGATABLE } from './common';
+
+export const hoveredRow$ = new BehaviorSubject<any | null>(null);
+
+const BAR_COLOR = euiPaletteColorBlind()[0];
+const BAR_COLOR_BLUR = euiPaletteColorBlind(2)[10];
+const MAX_CHART_COLUMNS = 20;
+
+const getXScaleType = (
+  kbnFieldType: KBN_FIELD_TYPES
+): 'ordinal' | 'time' | 'linear' | undefined => {
+  switch (kbnFieldType) {
+    case KBN_FIELD_TYPES.BOOLEAN:
+    case KBN_FIELD_TYPES.IP:
+    case KBN_FIELD_TYPES.STRING:
+      return 'ordinal';
+    case KBN_FIELD_TYPES.DATE:
+      return 'time';
+    case KBN_FIELD_TYPES.NUMBER:
+      return 'linear';
+  }
+};
+
+const getFieldType = (schema: EuiDataGridColumn['schema']): KBN_FIELD_TYPES | undefined => {
+  if (schema === NON_AGGREGATABLE) {
+    return undefined;
+  }
+
+  let fieldType: KBN_FIELD_TYPES;
+
+  switch (schema) {
+    case 'datetime':
+      fieldType = KBN_FIELD_TYPES.DATE;
+      break;
+    case 'numeric':
+      fieldType = KBN_FIELD_TYPES.NUMBER;
+      break;
+    case 'boolean':
+      fieldType = KBN_FIELD_TYPES.BOOLEAN;
+      break;
+    case 'json':
+      fieldType = KBN_FIELD_TYPES.OBJECT;
+      break;
+    default:
+      fieldType = KBN_FIELD_TYPES.STRING;
+  }
+
+  return fieldType;
+};
+
+export const fetchChartsData = async (
```

Comment:
When trying to show charts for a classification job (on `edibility`), I see the following error, and the charts fail to display:

![image](https://user-images.githubusercontent.com/7405507/84653183-da92b880-af04-11ea-8708-c58d19cfa8f9.png)

Reply:
I am still seeing this error when creating a transform with the `mushroom` index and trying to show the charts:

![image](https://user-images.githubusercontent.com/7405507/84883476-76006680-b088-11ea-8312-6eed085d90b5.png)

Reply:
Fixed in aaf6bdf. This also includes a tweak to avoid unnecessary remote calls when there are e.g. no numeric and/or string based fields or no chartable fields at all.

Reply:
Fixed in aaf6bdf.